crates/astria-sequencer/src/fees/action.rs

```diff
@@ -137,98 +137,290 @@ impl ActionHandler for FeeAssetChange {
+            new_fees,
+            new_fee_change,
+            "ibc_relayer_change",
+        )
+        .await;
+    }
+
+    #[tokio::test]
+    async fn fee_asset_change_fee_change_action_executes_as_expected() {
+        let (initial_fees, initial_fee_change, new_fees, new_fee_change) =
+            get_default_fees_and_fee_changes!(FeeAssetChange);
+        test_fee_change_action::<FeeAssetChange, _>(
+            initial_fees,
+            initial_fee_change,
+            new_fees,
+            new_fee_change,
+            "fee_asset_change",
+        )
+        .await;
+    }
+
+    #[tokio::test]
+    async fn fee_change_fee_change_action_executes_as_expected() {
+        let (initial_fees, initial_fee_change, new_fees, new_fee_change) =
+            get_default_fees_and_fee_changes!(FeeChange);
+        test_fee_change_action::<FeeChange, _>(
+            initial_fees,
+            initial_fee_change,
+            new_fees,
+            new_fee_change,
+            "fee_change",
+        )
+        .await;
+    }
+
+    #[tokio::test]
+    async fn sudo_address_change_fee_change_action_executes_as_expected() {
+        let (initial_fees, initial_fee_change, new_fees, new_fee_change) =
+            get_default_fees_and_fee_changes!(SudoAddressChange);
+        test_fee_change_action::<SudoAddressChange, _>(
+            initial_fees,
+            initial_fee_change,
+            new_fees,
+            new_fee_change,
+            "sudo_address_change",
+        )
+        .await;
+    }
+
+    #[tokio::test]
+    async fn ibc_sudo_change_fee_change_action_executes_as_expected() {
+        let (initial_fees, initial_fee_change, new_fees, new_fee_change) =
+            get_default_fees_and_fee_changes!(IbcSudoChange);
+        test_fee_change_action::<IbcSudoChange, _>(
+            initial_fees,
+            initial_fee_change,
+            new_fees,
+            new_fee_change,
+            "ibc_sudo_change",
+        )
+        .await;
+    }
+
+    async fn test_fee_change_action<F: FeeHandler<FeeComponents = T>, T>(
+        initial_fees: T,
+        initial_fee_change: FeeChange,
+        new_fees: T,
+        new_fee_change: FeeChange,
+        action_name: &str,
+    ) where
+        T: FeeComponents + std::fmt::Debug + std::cmp::PartialEq<T>,
+    {
+        let storage = cnidarium::TempStorage::new().await.unwrap();
+        let snapshot = storage.latest_snapshot();
+        let mut state = cnidarium::StateDelta::new(snapshot);
+
+        // Put the context to enable the txs to execute.
+        state.put_transaction_context(TransactionContext {
+            address_bytes: [1; 20],
+            transaction_id: TransactionId::new([0; 32]),
+            source_action_index: 0,
+        });
+        state.put_sudo_address([1; 20]).unwrap();
+
+        assert!(
+            F::fee_components(&state)
+                .await
+                .unwrap_or_else(|_| panic!("should not error fetching unstored {action_name} fees"))
```

Comment:
```suggestion
                .expect("should not error fetching unstored {action_name} fees")
```

same for other `unwrap_or_else` with `panic!`

Reply:
Done in 3a71450